config/v1/types_infrastructure.go

```diff
@@ -52,10 +52,16 @@ type InfrastructureStatus struct {
 	// For more info: https://github.com/etcd-io/etcd/blob/329be66e8b3f9e2e6af83c123ff89297e49ebd15/Documentation/op-guide/clustering.md#dns-discovery
 	EtcdDiscoveryDomain string `json:"etcdDiscoveryDomain"`
 
-	// apiServerURL is a valid URL with scheme(http/https), address and port.
-	// apiServerURL can be used by components like kubelet on machines, to contact the `apisever`
-	// using the infrastructure provider rather than the kubernetes networking.
+	// apiServerURL is a valid URI with scheme(http/https), address and
+	// port.  apiServerURL can be used by components like the web console
+	// to tell users where to find the Kubernetes API.
 	APIServerURL string `json:"apiServerURL"`
+
+	// apiServerInternalURL is a valid URI with scheme(http/https),
+	// address and port.  apiServerInternalURL can be used by in-cluster
+	// components like the machine-config operator to tell the kubelet
```

Comment:
nit, we should bee the `can be used by components like kubelet on machines, to contact the apisever using the infrastructure provider rather than the kubernetes networking.`

Reply:
> nit, we should keep the `can be used by components like kubelet on machines, to contact the apisever using the infrastructure provider rather than the kubernetes networking.`

I'd rather use "Kubernetes API" or "Kubernetes API server" instead of "`apiserver`".  Are you ok with that?  And can we mention Services instead of "kubernetes networking"?  I'd also like to name-drop the machine-config operator so there's less head-scratching about how the kubelet gets this information.  Rolling that all together with the previous wording, how about something like:

> apiServerInternalURL is intended for in-cluster components like the kubelet (via the machine-config operator) which need access to the Kubernetes API before their node has had time to provision Service networking.

How does that look?  Further tweak ideas?

Reply:
> apiServerInternalURL is intended for in-cluster components like the kubelet (via the machine-config operator) which need access to the Kubernetes API before their node has had time to provision Service networking.

`node has had time to provision Service networking`

you are making things more complex and `contact kubernetes API through somethings other than kubernetes networking` is the reason we have this.. whatever makes that clear. 

Services vs kubernetes networking: i prefer the latter.

via mco is not required in api docs.

Reply:
> you are making things more complex...

Ah well, my proposed changes seemed clear to me ;).  But I've pushed a33660b8 -> fdc055ce getting very close to your wording, just differing in "components like kubelet on machines" -> "components like kubelets", "`apiserver`", "Kubernetes API server", and "the kubernetes networking" -> "Kubernetes networking".  How's that look?